test/runtests.jl

```diff
@@ -290,6 +290,39 @@ make3() = (1.0, 2.0, 3.0)
 
 end
 
+@testset "Deferred and deferred thunk" begin
+    function square!(A)
+        A .*= A
```

Comment:
Can you simplify this test. For example, to have it be

```
function tapetest(A)
   return A[0] * A[1]
end
```

This should have a tape type of two float64's, rather than the complex and julia-version specific tape type from a broadcast

Reply:
Done.